### PR TITLE
plan: Choose left table as build table for semi join when left one is smaller for MPPTask

### LINKED ISSUE #43062: Choose left table as build side for semi/anti join when left table is smaller
## Enhancement
https://github.com/pingcap/tiflash/issues/7280

### PR DESCRIPTION
**Pull request by yibin87**
### What problem does this PR solve?

Issue Number: close #43062  

https://github.com/pingcap/tiflash/issues/7280

Problem Summary:

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
Improve semi join and anti semi join performance of MPPTasks when left table has smaller size than right table.
```

### CODE REVIEW

File: planner/core/exhaust_physical_plans.go [hunk 1]
```diff
@@ -2387,6 +2387,9 @@ func (p *LogicalJoin) tryToGetMppHashJoin(prop *property.PhysicalProperty, useBC
 		}
 	} else if p.JoinType.IsSemiJoin() {
 		preferredBuildIndex = 1
+		if p.children[1].statsInfo().Count() > p.children[0].statsInfo().Count() {
+			preferredBuildIndex = 0
+		}
 		fixedBuildSide = true
```
Comment (SeaRise) [inline planner/core/exhaust_physical_plans.go:2393]: Looks like `fixedBuildSide` is unnecessary.

Comment (yibin87) [reply planner/core/exhaust_physical_plans.go:2393]: Yeah, make sense, will remove it.

**Review by SeaRise — APPROVED** [review]

File: planner/core/exhaust_physical_plans.go [hunk 2]
```diff
@@ -2387,7 +2387,11 @@ func (p *LogicalJoin) tryToGetMppHashJoin(prop *property.PhysicalProperty, useBC
 		}
 	} else if p.JoinType.IsSemiJoin() {
 		preferredBuildIndex = 1
-		fixedBuildSide = true
+		// MPPOuterJoinFixedBuildSide default value is false
+		// use MPPOuterJoinFixedBuildSide here as a way to disable using left table as build side
+		if !p.ctx.GetSessionVars().MPPOuterJoinFixedBuildSide && p.children[1].statsInfo().Count() > p.children[0].statsInfo().Count() {
+			preferredBuildIndex = 0
+		}
```
Comment (windtalker) [inline planner/core/exhaust_physical_plans.go:2394]: Still need to set `fixedBuildSide` to `true` if `p.ctx.GetSessionVars().MPPOuterJoinFixedBuildSide` is true?

**Review by windtalker — APPROVED** [review]
LGTM

**Pull request merged**